### PR TITLE
[Maintenance] Make filters "title" only used for display

### PR DESCRIPTION
**Pull request by jotak**
This triggered some refactoring:
- New ActiveFilter "title" field
- Some simplification in StatefulFilters
- More typing: "groupBy" function, LabelFilters explicit call to avoid 'any'
- Move labelFilter type to CommonFilters.ts

Fixes https://github.com/kiali/kiali/issues/2785

### CODE REVIEW

**Comment by jotak** [comment]
Note: this is only refactoring, no visual change expected

**Comment by jotak** [comment]
@prachiyadav : for testing, basically the scope for non-regression is about filtering in the various list pages + overview

File: src/components/FilterList/FilterHelper.ts [hunk 1]
```diff
@@ -63,17 +63,17 @@ export const filtersMatchURL = (filterTypes: FilterType[], filters: ActiveFilter
...
   const urlParams = new URLSearchParams(history.location.search);
   filterTypes.forEach(filter => {
     const values = urlParams.getAll(filter.id);
     if (values.length > 0) {
-      const existing = fromURL.get(filter.title) || [];
-      fromURL.set(filter.title, existing.concat(values));
+      const existing = fromURL.get(filter.id) || [];
+      fromURL.set(filter.id, existing.concat(values));
     }
   });
```
Comment (aljesusg) [inline src/components/FilterList/FilterHelper.ts:78]: Yep this is the problem that I commented you about that filter used title instead I'd

File: src/helpers/LabelFilterHelper.ts [hunk 2]
```diff
@@ -69,8 +68,6 @@ const andLabelOperation = (labels: { [key: string]: string }, filters: string[])
         value.split(',').map(val => {
           // Split label values for serviceList Case where we can have multiple values for a label
           if (!labels[key].split(',').some(labelVal => labelVal.trim().startsWith(val.trim()))) {
-            console.log(labels[key].split(','));
-            console.log(val);
             filterOkForLabel = false;
           }
           return undefined;
```
Comment (aljesusg) [inline src/helpers/LabelFilterHelper.ts:73]: Omg :'(

**Pull request merged**